central/resourcecollection/datastore/datastore.go

```diff
@@ -37,15 +37,14 @@ type DataStore interface {
 }
 
 // New returns a new instance of a DataStore.
-func New(storage store.Store, indexer index.Indexer, searcher search.Searcher) DataStore {
+func New(storage store.Store, indexer index.Indexer, searcher search.Searcher) (DataStore, error) {
 	ds := &datastoreImpl{
 		storage:  storage,
 		indexer:  indexer,
 		searcher: searcher,
 		graph:    nil,
 	}
-
-	return ds
+	return ds, ds.initGraph()
```

Comment:
make sure `nil` is returned in case of error.
```suggestion
	if err := ds.initGraph(); err != nil {
	  return nil, err
	}
	return ds, nil
```